### DISCUSSION TITLE
New approach for documentation

### DISCUSSION THREAD

**Discussion by TheRealKeto** [ProcursusTeam/Procursus#853] [Show and tell]
I don't know this works... testing, testing, 1, 2...

Recently, [a PR was made by Katri that added a subfolder](https://github.com/ProcursusTeam/Procursus/pull/844) containing documentation about the build system, with a possible way to push that to the Wiki page whenever new changes were made. I came up with the idea of having that be handled separately from the Procursus repository, and I've just finished that.

The implementation of actually pushing changes to the Wiki section of this repository hasn't been implemented yet, since much needs to be discussed about the actual wiki repository, but (much like Procursus) contributions to add that can be made; the wiki repository can be found [here](https://github.com/TheRealKeto/procursus-wiki)... the conversation is open to anyone who'd like to give their opinion about this new approach on documentation for the project (which I feel obligated to run), as well as further ideas to make it better.

**Comment by crystall1nedev** [comment]
What if you chose to have a VuePress website for the wiki? You get the same flexibility with Markdown and the ability for outside sources to make it better, while also keeping it out of the way and easier to use. We (PojavLauncherTeam) chose this path and it has served us well, ViePress is completely customizable and easy to use (the only real hard part is config.js, but once you’ve learned what’s what it’s not hard.)

VuePress requires NodeJS to work and has a starter script. You can also check [our website](https://github.com/PojavLauncherTeam/pojavlauncherteam.github.io) for a more complete example.

**Reply by CRKatri** [reply]
I don't think we need a full website, the GitHub wiki serves well so far.

**Reply by crystall1nedev** [reply]
True true

**Reply by TheRealKeto** [reply]
While full website might be good in some cases, it's not in this one; the implementation that I (as well as Katri) had was to use a Github Action/Workflow; a full fledged website runs the risk of hosting and making sure that any new changes gets deployed, and that's too tedious for right now.

**Reply by crystall1nedev** [reply]
See the thing is that I *was* talking about VuePress using GitHub Pages. All you would need to do is make changes to the documentation, push, and then a GitHub workflow similar to ours pushes the results to the pages branch. Themes, assets, and everything else is super easy to deal with too.
